app/grandchallenge/reader_studies/views.py

```diff
@@ -1476,8 +1486,28 @@ def create_civ_for_value(
             )
             # Add the provided civ to the current display set
             instance.values.add(new_value)
+        elif not new_value:
+            # if no new value is provided (user selects '---' in dropdown)
+            # delete old CIV
+            assigned_civs.append(current_civ)
         return instance, assigned_civs
 
+    def create_civ(self, instance, ci_slug, new_value, assigned_civs):
+        ci = ComponentInterface.objects.get(slug=ci_slug)
+        current_civ = instance.values.filter(interface=ci).first()
+        if ci.is_json_kind and not ci.requires_file:
+            return self.create_civ_for_value(
+                instance, ci, current_civ, new_value, assigned_civs
+            )
+        elif ci.is_image_kind:
+            return self.create_civ_for_image(
+                instance, ci, current_civ, new_value, assigned_civs
+            )
+        elif ci.requires_file:
+            return self.create_civ_for_file(
+                instance, ci, current_civ, new_value, assigned_civs
+            )
```

Comment:
Should we catch anything in a final `else` block here?

Reply:
Good idea, yes. At this point, the ifs cover all cases, but that could of course change in the future.